framework/packages/abstract-client/src/application.rs

```diff
@@ -32,7 +32,7 @@ impl<Chain: CwEnv, M> DerefMut for Application<Chain, M> {
     }
 }
 
-impl<Chain: CwEnv, M: RegisteredModule> Application<Chain, M> {
+impl<Chain: CwEnv, M: RegisteredModule + ContractInstance<Chain>> Application<Chain, M> {
```

Comment:
If you only need that bound in the `authorize_on_adapters` you can create a new impl with that bound added to it and leave this impl without it.

Alternatively we could add a supertrait that overwraps these traits so the signatures are smaller.

Reply:
Fixed